common/storage/sql/sqlite3/storage.c

```diff
@@ -639,3 +639,91 @@ retcode_t iota_stor_milestone_exist(const connection_t* const conn,
   }
   return RC_OK;
 }
+
+/*
+ * State diff operations
+ */
+
+retcode_t iota_stor_state_diff_store(const connection_t* const conn,
+                                     uint64_t index, state_map_t* diff) {
+  retcode_t ret = RC_OK;
+  char const* err_msg = 0;
+  sqlite3_stmt* sqlite_statement = 0;
+  char statement[MILESTONE_MAX_UPDATE_STATEMENT_SIZE];
+  size_t size = 0;
+  byte_t* bytes = NULL;
+
+  if ((ret = iota_statement_state_diff_store(
+           index, statement, MILESTONE_MAX_UPDATE_STATEMENT_SIZE))) {
+    return ret;
+  }
+
+  if ((ret = prepare_statement((sqlite3*)conn->db, &sqlite_statement, statement,
+                               &err_msg))) {
+    return ret;
+  }
+
+  size = iota_state_diff_serialized_size(diff);
+  bytes = calloc(size, sizeof(byte_t));
+  if ((ret = iota_state_diff_serialize(diff, bytes)) != RC_OK) {
+    goto done;
+  }
+  int rc = sqlite3_bind_blob(sqlite_statement, 1, bytes, size, NULL);
+
+  if (rc != SQLITE_OK) {
+    log_error(SQLITE3_LOGGER_ID,
+              "Failed in binding, sqlite3 code is: %" PRIu64 "\n", rc);
+    ret = RC_SQLITE3_FAILED_BINDING;
+    goto done;
+  }
+
+  if ((ret = execute_statement_store(sqlite_statement))) {
+    goto done;
+  }
+
+done:
```

Comment:
This is missing a call to `sqlite3_finalize`: https://www.sqlite.org/c3ref/finalize.html

Reply:
It's done in `execute_statement_store`

Reply:
Well, it probably shouldn't be done in `execute_statement_store `.
I'm doing a storage refactoring in another branch rn.

Reply:
Fixed by #398

Reply:
Thanks :)